### ISSUE TITLE
HLS crashes when triggering module names completion

### ISSUE THREAD

**Issue by berberman** [haskell/haskell-language-server#1365]
### HLS verson:
```
haskell-language-server version: 0.9.0.0 (GHC: 8.10.3) (PATH: /home/berberman/.cabal/store/ghc-8.10.3/haskell-language-server-0.9.0.0-0a1f130421b85b97d7d539f51f666b79f189459c3d06c11eb1bfa9e18e0dbb5d/bin/haskell-language-server)
Tool versions found on the $PATH
cabal:          3.2.0.0
stack:          Not found
ghc:            8.10.3
```
### Screenshot
![Peek](https://user-images.githubusercontent.com/26041945/107872492-9f0e1b00-6ee5-11eb-9bfc-4d0d13f68e50.gif)

relevant log from output:

```
haskell-language-server: haskell-language-server: panic! (the 'impossible' happened)
  (GHC version 8.10.3:
	ModOrigin: hidden module redefined

Please report this as a GHC bug:  https://www.haskell.org/ghc/reportabug
```

I think it's related to a reported ghc bug: https://gitlab.haskell.org/ghc/ghc/-/issues/16996 -- HLS crashes only when loading projects where we are not able to use module completions in ghci, as discussed in the gitlab issue.

**Comment by pepeiborra** [comment]
This was also discussed in #786

**Comment by jneira** [comment]
@berberman this can be closed by #1367 or we could do something to improve state of things?
I guess it is blocked on the mentioned ghc bug

**Comment by michaelpj** [comment]
I'm assuming this is fixed
